test/AWS.Deploy.CLI.IntegrationTests/BeanstalkBackwardsCompatibilityTests/ExistingWindowsEnvironment/ServerModeTests.cs

```diff
@@ -0,0 +1,132 @@
+            try
+            {
+                var baseUrl = $"http://localhost:{portNumber}/";
+                var restClient = new RestAPIClient(baseUrl, httpClient);
+
+                await WaitTillServerModeReady(restClient);
+
+                var startSessionOutput = await restClient.StartDeploymentSessionAsync(new StartDeploymentSessionInput
+                {
+                    AwsRegion = "us-west-2",
+                    ProjectPath = projectPath
+                });
+
+                var sessionId = startSessionOutput.SessionId;
+                Assert.NotNull(sessionId);
+
+                var existingDeployments = await restClient.GetExistingDeploymentsAsync(sessionId);
+                var existingDeployment = existingDeployments.ExistingDeployments.First(x => string.Equals(_fixture.EnvironmentName, x.Name));
+
+                Assert.Equal(_fixture.EnvironmentName, existingDeployment.Name);
+                Assert.Equal(BEANSTALK_ENVIRONMENT_RECIPE_ID, existingDeployment.RecipeId);
+                Assert.Null(existingDeployment.BaseRecipeId);
+                Assert.False(existingDeployment.IsPersistedDeploymentProject);
+                Assert.Equal(_fixture.EnvironmentId, existingDeployment.ExistingDeploymentId);
+                Assert.Equal(DeploymentTypes.BeanstalkEnvironment, existingDeployment.DeploymentType);
+
+                var signalRClient = new DeploymentCommunicationClient(baseUrl);
+                await signalRClient.JoinSession(sessionId);
+
+                var logOutput = new StringBuilder();
+                AWS.Deploy.CLI.IntegrationTests.ServerModeTests.RegisterSignalRMessageCallbacks(signalRClient, logOutput);
+
+                await restClient.SetDeploymentTargetAsync(sessionId, new SetDeploymentTargetInput
+                {
+                    ExistingDeploymentId = _fixture.EnvironmentId
+                });
+
+                await restClient.StartDeploymentAsync(sessionId);
+
+                await WaitForDeployment(restClient, sessionId);
+
+                Assert.True(logOutput.Length > 0);
+                var successMessagePrefix = $"The Elastic Beanstalk Environment {_fixture.EnvironmentName} has been successfully updated";
+                var deployStdOutput = logOutput.ToString().Split(Environment.NewLine);
+                var successMessage = deployStdOutput.First(line => line.Trim().StartsWith(successMessagePrefix));
+                Assert.False(string.IsNullOrEmpty(successMessage));
+
+                var expectedVersionLabel = successMessage.Split(" ").Last();
+                Assert.True(await _fixture.EBHelper.VerifyEnvironmentVersionLabel(_fixture.EnvironmentName, expectedVersionLabel));
+            }
+            finally
+            {
+                cancelSource.Cancel();
+            }
+        }
+
+        private async Task<DeploymentStatus> WaitForDeployment(RestAPIClient restApiClient, string sessionId)
```

Comment:
There are three identical implementations of some of these utilities in this branch! Please reuse instead of redefining.

Reply:
Done. I did this for multiple server mode functions that we were not reusing.